subxt/src/client/lightclient/background.rs

```diff
@@ -0,0 +1,428 @@
+
+                let result = self.client.json_rpc_request(request, self.chain_id);
+                if let Err(err) = result {
+                    tracing::warn!(
+                        target: LOG_TARGET,
+                        "Cannot send RPC request to lightclient {:?}",
+                        err.to_string()
+                    );
+                    let sender = self
+                        .requests
+                        .remove(&id)
+                        .expect("Channel is inserted above; qed");
+
+                    // Send the error back to frontend.
+                    if sender
+                        .send(Err(LightClientError::Request(err.to_string())))
+                        .is_err()
+                    {
+                        tracing::warn!(
+                            target: LOG_TARGET,
+                            "Cannot send RPC request error to id{:?}",
+                            id
+                        );
+                    }
+                }
+            }
+            FromSubxt::Subscription {
+                method,
+                params,
+                sub_id,
+                sender,
+            } => {
+                // For subscriptions we need to make a plain RPC request to the subscription method.
+                // The server will return as a result the subscription ID.
+                let id = self.next_id();
+                let request = format!(
+                    r#"{{"jsonrpc":"2.0","id":"{}", "method":"{}","params":{}}}"#,
+                    id, method, params
+                );
+                let id = id.to_string();
+
+                self.id_to_subscription.insert(id.clone(), (sub_id, sender));
+
+                let result = self.client.json_rpc_request(request, self.chain_id);
+                if let Err(err) = result {
+                    tracing::warn!(
+                        target: LOG_TARGET,
+                        "Cannot send RPC request to lightclient {:?}",
+                        err.to_string()
+                    );
+                    let (sub_id, _) = self
+                        .id_to_subscription
+                        .remove(&id)
+                        .expect("Channels are inserted above; qed");
+
+                    // Send the error back to frontend.
+                    if sub_id
+                        .send(Err(LightClientError::Request(err.to_string())))
+                        .is_err()
+                    {
+                        tracing::warn!(
+                            target: LOG_TARGET,
+                            "Cannot send RPC request error to id{:?}",
+                            id
+                        );
+                    }
+                }
+            }
+        };
+    }
+
+    /// Parse the response received from the light client and sent it to the appropriate user.
+    async fn handle_rpc_response(&mut self, response: String) {
```

Comment:
looks like this function doesn't have to be async

Reply:
The sender channel for the subscription returns a future